dds_web/api/marshmallows.py

```diff
@@ -80,3 +94,194 @@ def return_items(self, data, **kwargs):
+            raise marshmallow.ValidationError("currently not creating invites for superadmins")
+
+        # Create invite
+        new_invite = models.Invite(**{"email": data.get("email"), "role": data.get("role")})
+
+        # Append invite to unit if applicable
+        if new_invite.role in ["Unit Admin", "Unit Personnel"]:
+            auth.current_user().unit.invites.append(new_invite)
+        else:
+            db.session.add(new_invite)
+
+        db.session.commit()
+
+        return new_invite
+
+
+class InviteSchema(marshmallow.Schema):
+
+    unit_id = marshmallow.fields.Integer(required=True)
+    email = marshmallow.fields.Email(required=True, validate=marshmallow.validate.Length(max=254))
+    role = marshmallow.fields.String(required=True, validate=marshmallow.validate.Length(max=20))
+
+
+class NewUserSchema(marshmallow.Schema):
+    """Schema for NewUser endpoint"""
+
+    # TODO: Look through and match to db
+    username = marshmallow.fields.String(
+        required=True,
+        validate=marshmallow.validate.Regexp(
+            regex="^[a-zA-Z0-9_]{6,25}$",
+            # username must be between 6-25 characters long and contain a-z, A-Z, 0-9
+        ),
+    )
+    password = marshmallow.fields.String(
+        required=True,
+        validate=marshmallow.validate.Regexp(
+            regex="^(?=[^a-z\n]*[a-z])(?=[^A-Z\n]*[A-Z])(?=[^\d\n]*\d)^[\S]{10,64}$",
+            # Password must contain one or more uppercase, lowercase and digit characters.
+            # It may contain special characters (not enforced), as they might be hard to type depending on keyboard layout (e.g. on smartphones)
+            # Length of the password must be between 10 and 64 characers
+        ),
+    )
+    email = marshmallow.fields.Email(required=True, validate=marshmallow.validate.Email())
+    name = marshmallow.fields.String(required=True, validate=marshmallow.validate.Length(max=255))
+
+    class Meta:
+        """Exclude unknown fields e.g. csrf etc that are passed with form"""
+
+        unknown = marshmallow.EXCLUDE
+
+    @marshmallow.validates("email")
+    def verify_new_email(self, value):
+        """Verify that the email is not used in the system already."""
+
+        if email_in_db(email=value):
+            raise marshmallow.ValidationError(
+                message=f"The email '{value}' is already registered to an existing user."
+            )
+
+    @marshmallow.validates_schema(skip_on_field_errors=True)
+    def verify_and_get_invite(self, data, **kwargs):
+        """Verifies that the email is in the invite table and in that case saves the invite info."""
+
+        invite = models.Invite.query.filter(
+            models.Invite.email == sqlalchemy.func.binary(data.get("email"))
+        ).one_or_none()
+        if not invite:
+            raise ddserr.InviteError
+
+        data["invite"] = invite
+
+    @marshmallow.post_load
+    def make_user(self, data, **kwargs):
```

Comment:
I kind of agree, but I will leave this for now. Since I'm in the process of adding marshmallows to other endpoints too I can have a look then what makes the most sense.